utils/index.sh

```diff
@@ -0,0 +1,132 @@
+        "ci_system" : "'$ci'",
+        "uuid" : "'$UUID'",
+        "release_stream": "'$RELEASE_STREAM'",
+        "platform": "'$platform'",
+        "master_count": '$masters',
+        "worker_count": '$workers',
+        "infra_count": '$infra',
+        "master_type": "'$master_type'",
+        "worker_type": "'$worker_type'",
+        "infra_type": "'$infra_type'",
+        "total_count": '$all',
+        "cluster_name": "'$cluster_name'",
+        "cluster_version": "'$cluster_version'",
+        "network_type": "'$network_type'",
+        "build_tag": "'$task_id'",
+        "node_name": "'$HOSTNAME'",
+        "job_status": "'$state'",
+        "build_url": "'$build_url'",
+        "upstream_job": "'$dag_id'",
+        "upstream_job_build": "'$dag_run_id'",
+        "execution_date": "'$execution_date'",
+        "job_duration": "'$duration'",
+        "start_date": "'"$start_date"'",
+        "end_date": "'"$end_date"'",
+        "timestamp": "'"$start_date"'"
+        }' "$url"
+}
+
+set_duration(){
+    start_date="$1"
+    end_date="$2"
+    if [[ -z $start_date ]]; then
+        start_date=$end_date
+    fi
+
+    if [[ -z $start_date || -z $end_date ]]; then
+        duration=0
+    else
+        end_ts=$(date -d "$end_date" +%s)
+        start_ts=$(date -d "$start_date" +%s)
+        duration=$(( $end_ts - $start_ts ))
+    fi
+}
+
+
+index_tasks(){
+    if [[ -z $PROW_JOB_ID ]]; then
+        task_states=$(AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR  airflow tasks states-for-dag-run $dag_id $execution_date -o json)
+        task_json=$( echo $task_states | jq -c ".[] | select( .task_id == \"$TASK\")")
+        state=$(echo $task_json | jq -r '.state')
+        task_id=$(echo $task_json | jq -r '.task_id')
+
+        if [[ $task_id == "$AIRFLOW_CTX_TASK_ID" || $task_id == "cleanup" ]]; then
+            echo "Index Task doesn't index itself or cleanup step, skipping."
+        else
+            start_date=$(echo $task_json | jq -r '.start_date')
+            end_date=$(echo $task_json | jq -r '.end_date')
+            set_duration "$start_date" "$end_date"
+            encoded_execution_date=$(python3 -c "import urllib.parse; print(urllib.parse.quote(input()))" <<< "$execution_date")
+            build_url="${airflow_base_url}/task?dag_id=${dag_id}&task_id=${task_id}&execution_date=${encoded_execution_date}"
+            index_task "$ES_SERVER/$ES_INDEX/_doc/$dag_id%2F$dag_run_id%2F$task_id"
+        fi
+    else
+        task_id=$BUILD_ID
+        dag_id=$JOB_NAME
+        dag_run_id=$PROW_JOB_ID
+        state=$PROW_JOB_STATUS
+        build_url="${prow_base_url}/${dag_id}/${task_id}"
+        set_duration "$PROW_JOB_START" "$PROW_JOB_END"
+        index_task "$ES_SERVER/$ES_INDEX/_doc/$dag_id%2F$dag_run_id%2F$task_id"
+    fi
+
+}
+
+# Defaults
+if [[ -z $ES_SERVER ]]; then
+  echo "Elastic server is not defined, please check"
+  help
```

Comment:
help is not defined

Reply:
Thanks for catching that. Removing it as it is not defined.